sparse-array/src/fast.rs

```diff
@@ -0,0 +1,187 @@
+///! Wrapper for sparse arrays that doesn't delete anything from the memory
+/// unless `shrink` is called.
+
+use std::sync::Arc;
+use crate::{
+    bitmap::BitmapIndex,
+    SparseArray,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct FastSparseArray<V> {
+    index: BitmapIndex,
+    data: Arc<SparseArray<V>>,
```

Comment:
I don't get the reason for the presence of the `Arc`?

Reply:
Initially I wanted to use `Rc` to share this field between instances of `FastSparseArray`, but apparently it is used by different threads in the UTXO module.

The idea behind sharing this data is making faster clones, I discussed this one with Vincent.

Reply:
ok got it